contracts/Strategy.sol

```diff
@@ -53,80 +222,620 @@ contract Strategy is BaseStrategy {
+            kashiPairs.pop();
+            return;
+        }
+
+        revert("kashiPair not found");
+    }
+
+    function adjustKashiPairRatios(uint256[] calldata _ratios)
+        external
+        onlyAuthorized
+    {
+        // length of ratios must match number of pairs
+        require(_ratios.length == kashiPairs.length);
+
+        uint256 totalRatio;
+
+        for (uint256 i = 0; i < kashiPairs.length; i++) {
+            // We must accrue all pairs to ensure we get an accurate estimate of assets
+            accrueInterest(i);
+            totalRatio += _ratios[i];
+        }
+
+        require(totalRatio == MAX_BPS); //ratios must add to 10000 bps
+
+        uint256 wantBalance = balanceOfWant();
+        if (wantBalance > dustThreshold) {
+            depositInBento(wantBalance);
+        }
+
+        uint256 totalAssets = estimatedTotalAssets();
+        uint256[] memory kashiPairsIncreasedAllocation =
+            new uint256[](kashiPairs.length);
+
+        for (uint256 i = 0; i < kashiPairs.length; i++) {
+            uint256 pairTotalAssets =
+                bentoSharesToWant(
+                    kashiFractionToBentoShares(
+                        kashiPairs[i].kashiPair,
+                        kashiFractionTotal(i),
+                        true
+                    ),
+                    true
+                );
+            uint256 targetAssets = (_ratios[i] * totalAssets) / MAX_BPS;
+            if (targetAssets < pairTotalAssets) {
+                uint256 toLiquidate = pairTotalAssets.sub(targetAssets);
+                liquidateKashiPair(i, wantToBentoShares(toLiquidate, true));
+            } else if (targetAssets > pairTotalAssets) {
+                kashiPairsIncreasedAllocation[i] = targetAssets.sub(
+                    pairTotalAssets
+                );
+            }
+        }
+
+        for (uint256 i = 0; i < kashiPairs.length; i++) {
+            if (kashiPairsIncreasedAllocation[i] == 0) continue;
+
+            uint256 sharesInBento = sharesInBento();
+            uint256 sharesToAdd =
+                wantToBentoShares(kashiPairsIncreasedAllocation[i], true);
+
+            if (sharesToAdd > sharesInBento) {
+                sharesToAdd = sharesInBento;
+            }
+
+            depositInKashiPair(i, sharesToAdd);
+        }
+    }
+
+    function depositInKashiPair(uint256 kashiPairIndex, uint256 sharesToDeposit)
+        internal
+    {
+        transferBento(
+            address(kashiPairs[kashiPairIndex].kashiPair),
+            sharesToDeposit
+        );
+
+        uint256 depositedFraction =
+            kashiPairs[kashiPairIndex].kashiPair.addAsset(
+                address(this),
+                true,
+                sharesToDeposit
+            );
+
+        depositKashiInMasterChef(kashiPairIndex);
+    }
+
+    function depositKashiInMasterChef(uint256 kashiPairIndex) internal {
+        if (kashiPairs[kashiPairIndex].pid == 0) return;
+        masterChef.deposit(
```

Comment:
check for amount to deposit and don't deposit under a threshold to avoid depositing dust. does that make sense?